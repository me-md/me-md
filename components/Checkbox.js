import React, { useState } from 'react';
import { Dimensions, StyleSheet } from 'react-native';
import { Body, Card, Text } from 'native-base';
import { Feather } from '@expo/vector-icons';

const height = Dimensions.get('window').height;
const width = Dimensions.get('window').width;

export default function Checkbox() {
  const [checked, setChecked] = useState(false);

  return (
    <Card style={styles.card}>
      <Body style={styles.body}>
        <Feather
          name={checked ? 'x-square' : 'square'}
          size={24}
          color='black'
          onPress={() => {
            setChecked(!checked);
          }}
        />
        <Text>I have read and accept the Terms of Service</Text>
      </Body>
    </Card>
  );
}

const styles = StyleSheet.create({
  card: {
    flex: 0.2,
    justifyContent: 'space-evenly',
    width: width * 0.92
  },
  body: {
    alignItems: 'center',
    flex: 0.3,
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    width: '100%'
  }
});
