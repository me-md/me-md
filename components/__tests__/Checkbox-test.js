import React from 'react';
import renderer from 'react-test-renderer';

import Checkbox from '../Checkbox';

it(`should render correctly`, () => {
  const mockChangeButton = jest.fn();
  
  const tree = renderer.create(<Checkbox changeButton={mockChangeButton} />).toJSON();
  expect(tree).toMatchSnapshot();
});