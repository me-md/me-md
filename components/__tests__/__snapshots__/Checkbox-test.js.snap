// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderLeftWidth": 0.5,
          "borderRadius": 2,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        },
        Object {
          "flex": 0.45,
          "justifyContent": "space-evenly",
          "width": 690,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "width": "100%",
          },
        ]
      }
    >
      <Text />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
        uppercase={false}
      >
        I have read and accept the Terms of Service
      </Text>
    </View>
  </View>
</View>
`;
